### DISCUSSION TITLE
set visible Index when I change Tab

### DISCUSSION THREAD

**Discussion by minsukim-29cm** [asmyshlyaev177/react-horizontal-scrolling-menu#294] [Q&A] [answered]
<img width="839" alt="image" src="https://github.com/user-attachments/assets/631ee86c-e419-469d-835d-126df301db66">
<img width="788" alt="image" src="https://github.com/user-attachments/assets/c262ad12-7e6a-45e2-b4da-0a13120a9671">

Case.
1. change first Tab's visible element By arrow
2. click the Tab(해외주식)
3. It shows SAME index with previous Index

There are 2 tabIndex ("국내주식", "해외주식").
And It seems that they share same Index in scroll.
I want to maintain seperate visible Index.

**Comment by asmyshlyaev177** [comment, marked as answer]
Hi @durumi99 , can pass `apiRef` ref prop to menu, and reset state in `useEffect`, e.g. when selected tab changes.
https://codesandbox.io/p/sandbox/react-horizontal-scrolling-menu-v2-apiref-vdr0d?file=%2Fsrc%2Findex.tsx%3A37%2C14
![image](https://github.com/user-attachments/assets/1484a5a5-4a8e-4a3c-acaf-c790ee2e8fc3)

**Comment by asmyshlyaev177** [comment]
@durumi99  Another option is to try to pass current tab name as `key` prop to a menu:
```typescript
<ScrollMenu key={selectedTab}>
...
</ScrollMenu>

```
